source/frontend/src/features/leases/list/LeaseListView.test.tsx

```diff
@@ -171,6 +173,204 @@ describe('Lease and License List View', () => {
+  });
+
+  it('searches historical file number for PS', async () => {
+    setupMockSearch([
+      {
+        ...getEmptyLease(),
+        id: 1,
+        lFileNo: 'L-123-456',
+        programName: 'TRAN-IT',
+        tenants: [
+          {
+            ...getEmptyLeaseTenant(),
+            person: { ...getEmptyPerson(), firstName: 'Chester', surname: 'Tester' },
+          },
+        ],
+        fileProperties: [
+          {
+            ...getEmptyPropertyLease(),
+            property: {
+              ...getEmptyProperty(),
+              id: 123,
+              address: { ...getEmptyAddress(), streetAddress1: '123 mock st' },
+              pin: 123,
+              fileNumbers: [
+                {
+                  id: 1000,
+                  fileNumber: '0309-000',
+                  fileNumberTypeCode: {
+                    id: ApiGen_CodeTypes_HistoricalFileNumberTypes.PSNO.toString(),
+                    description: 'PS #',
+                    isDisabled: false,
+                    displayOrder: 3,
+                  },
+                  propertyId: 123,
+                  property: null,
+                  otherFileNumberType: null,
+                  isDisabled: false,
+                  rowVersion: 1,
+                  appCreateTimestamp: '',
+                  appLastUpdateTimestamp: '',
+                  appLastUpdateUserid: '',
+                  appCreateUserid: '',
+                  appLastUpdateUserGuid: '',
+                  appCreateUserGuid: ''
+                },
+              ]
+            },
+          },
+        ],
+      },
+    ]);
+    const { container, searchButton, findByText } = setup();
+
+    fillInput(container, 'searchBy', 'historical', 'select');
+    fillInput(container, 'historical', '0309-000');
+    await act(async () => userEvent.click(searchButton));
+
+    expect(getLeases).toHaveBeenCalledWith(
+      expect.objectContaining({
+        historical: '0309-000',
+        pinOrPid: '',
+        searchBy: 'historical',
+      }),
+    );
+
+    expect(await findByText(/0309-000;/i)).toBeInTheDocument();
+  });
+
+  it('searches historical file number for OTHER', async () => {
+    setupMockSearch([
+      {
+        ...getEmptyLease(),
+        id: 1,
+        lFileNo: 'L-123-456',
+        programName: 'TRAN-IT',
+        tenants: [
+          {
+            ...getEmptyLeaseTenant(),
+            person: { ...getEmptyPerson(), firstName: 'Chester', surname: 'Tester' },
+          },
+        ],
+        fileProperties: [
+          {
+            ...getEmptyPropertyLease(),
+            property: {
+              ...getEmptyProperty(),
+              id: 123,
+              address: { ...getEmptyAddress(), streetAddress1: '123 mock st' },
+              pin: 123,
+              fileNumbers: [
+                {
+                  id: 1000,
+                  fileNumber: '0309-999',
+                  fileNumberTypeCode: {
```

Comment:
nit: would be nice to create a function to create these file numbers, since more then half of the lines in the test are just setting up required fields in this model.